modules/economy-system/commands/shop.js

```diff
@@ -1,92 +1,42 @@
-            '%item%': item['value'],
-            '%price%': price,
-            '%role%': role.name
-        }, {ephemeral: !interaction.client.configurations['economy-system']['config']['publicCommandReplies']}));
-        interaction.client.logger.info(`[economy-system] ` + localize('economy-system', 'created-item', {
-            u: interaction.user.tag,
-            i: item['value']
-        }));
-        if (interaction.client.logChannel) interaction.client.logChannel.send(`[economy-system] ` + localize('economy-system', 'created-item', {
-            u: interaction.user.tag,
-            i: item['value']
-        }));
+        if (!checkPerms(interaction)) return;
+        await createShopItem(interaction);
+        shopMsg(interaction.client);
     },
     'buy': async function (interaction) {
-        const itemName = await interaction.options.get('item');
-        const item = await interaction.client.models['economy-system']['Shop'].findOne({
-            where: {
-                name: itemName['value']
-            }
-        });
-        if (!item) {
-            interaction.reply({
-                content: interaction.client.configurations['economy-system']['strings']['notFound'],
-                ephemeral: !interaction.client.configurations['economy-system']['config']['publicCommandReplies']
-            });
-        }
-        const user = await interaction.client.models['economy-system']['Balance'].findOne({
-            where: {
-                id: interaction.user.id
-            }
-        });
-        if (user.balance < item.price) return interaction.reply({
-            content: interaction.client.configurations['economy-system']['strings']['notEnoughMoney'],
-            ephemeral: !interaction.client.configurations['economy-system']['config']['publicCommandReplies']
-        });
-        await editBalance(interaction.client, interaction.user.id, 'remove', item.price);
-        await interaction.member.roles.add(item.role);
-        createleaderboard(interaction.client);
-        interaction.reply(embedType(interaction.client.configurations['economy-system']['strings']['buyMsg'], {'%item%': itemName['value']}, {ephemeral: !interaction.client.configurations['economy-system']['config']['publicCommandReplies']}));
-        interaction.client.logger.info(`[economy-system] ` + localize('economy-system', 'user-purchase', {
-            u: interaction.user.tag,
-            i: item['value'],
-            p: item['price']
-        }));
-        if (interaction.client.logChannel) interaction.client.logChannel.send(`[economy-system] ` + localize('economy-system', 'user-purchase', {
-            u: interaction.user.tag,
-            i: item['value'],
-            p: item['price']
-        }));
+        let name = await interaction.options.get('item-name');
+        if (name) name = name['value'];
+        let id = await interaction.options.get('item-id')['value'];
```

Comment:
This will cause a crash if item-id is not set. Use [`CommandInteractionOptionResolver.getString()`](https://discord.js.org/#/docs/discord.js/v13/class/CommandInteractionOptionResolver?scrollTo=getString) to simplify this